### PR TITLE
docs: update ``usage/caching``

### PR DESCRIPTION
**Pull request by JacobCoffee**
## Description

- updates `usage/caching`  as part of #3212 work
- followup of https://github.com/litestar-org/litestar/pull/3309

### CODE REVIEW

**Comment by JacobCoffee** [comment]
Good to go minus https://github.com/litestar-org/litestar/pull/3309#discussion_r1550874080

File: docs/usage/caching.rst [hunk 1]
```diff
@@ -15,59 +15,63 @@ expected. Litestar comes with a simple mechanism for caching:
...
-any :class:`Store <.stores.base.Store>`, for example :class:`RedisStore <.stores.redis.RedisStore>`:
+By default, caching will use the :class:`~.stores.memory.MemoryStore`, but it can be configured with
+any :class:`~.stores.base.Store`, for example :class:`~.stores.redis.RedisStore`:
 
 .. code-block:: python
+    :caption: Using Redis as the cache store.
 
-   from litestar.config.cache import ResponseCacheConfig
-   from litestar.stores.redis import RedisStore
+    from litestar.config.cache import ResponseCacheConfig
```
Comment (Alc-Alc) [inline docs/usage/caching.rst:69]: I think this is a remnant of an old pattern. IIUC the snippet in the docs would no longer work.

The following is an MCVE that shows how to use a custom store using the current interfaces.

```py
import asyncio
from litestar import Litestar, get
from litestar.config.response_cache import ResponseCacheConfig
from litestar.stores.redis import RedisStore


@get(cache=10)
async def something() -> str:
    await asyncio.sleep(1)
    return "something"


redis_store = RedisStore.with_client(url="redis://localhost", port=6379, db=0)
cache_config = ResponseCacheConfig(store="redis_backed_store")
app = Litestar(
    [something],
    stores={"redis_backed_store": redis_store},
    response_cache_config=cache_config,
)
```

Some may argue this is not good DX, I personally do not find this any difficult.

Comment (JacobCoffee) [reply docs/usage/caching.rst:69]: This is more evidence towards us needing to move any examples into `examples/` and have them tested with each PR so they dont go stale.

We dont have to show the whole app, we could `:linenos:` it, but we should really work on that

**Review by Alc-Alc — APPROVED** [review]
One could say the example docs were... *cached* 🙃

**Pull request merged**